### DISCUSSION TITLE
Edit mode

### DISCUSSION THREAD

**Discussion by sergey-tihon** [sergey-tihon/vscode-openxml-explorer#16] [Ideas]
For edit mode we can mount openxml zip archive as file system.

Types are here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/vscode/index.d.ts#L7026
Docs are here https://code.visualstudio.com/api/extension-guides/virtual-documents#file-system-api

```
  context.subscriptions.push(vscode.workspace.registerFileSystemProvider('zip', new ZipFileSystemProvider(), { isReadonly: true, isCaseSensitive: true }));
  context.subscriptions.push(vscode.commands.registerCommand('extension.mountZipFileSystem', (_uri: vscode.Uri) => {
    const uri = vscode.Uri.parse(`zip:/?${_uri}`);
    if (vscode.workspace.getWorkspaceFolder(uri) === undefined) {
      const name = vscode.workspace.asRelativePath(_uri, true);
      const index = vscode.workspace.workspaceFolders?.length || 0;
      const workspaceFolder: vscode.WorkspaceFolder = { uri, name, index };
      vscode.workspace.updateWorkspaceFolders(index, 0, workspaceFolder);
    }
  }));
```

https://github.com/TomasHubelbauer/vscode-zip-file-system/blob/master/src/extension.ts#L5-L12

I do not like experience, because it modify current workspace and we will have to show folders and all internal files like .rels

**Comment by sergey-tihon** [comment]
> Most of time I tweak custom Xml files used in content control binding (we built doc generation in a few projets).
I use either a manual process (unzip the package, edit the content and rezip the package). Or sometimes a chrome addon (https://chrome.google.com/webstore/detail/ooxml-tools/bjmmjfdegplhkefakjkccocjanekbapn)

https://twitter.com/stevebeauge/status/1405256904285294593

**Comment by stevebeauge** [comment]
Hi Sergey,

Documentation regarding xml data binding is quite poor, and only few people write about this.

Here's a few links though:
* https://docs.microsoft.com/en-us/office/client-developer/word/content-controls-in-word
* http://www.ericwhite.com/blog/
* https://www.docx4java.org/blog/2015/01/i-have-my-xml-now-what/

In our case, we often have to generate documents that contains content from backoffice apps.
The process consists in 
1. build an xml document that represent our data
2. inject it as custom XML part in template Word file
3. Bind content control and repeatable sections to the data in the xml part
4. SAve the template file in the document processing tool
5. When doc has to be generated, inject the Xml part in the word template and outputs to the user the doc

During the process of building the template, we often have to tweak the custom Xml file to adapt new cases, or to fix potential bugs.

As I said previously, we either unzip/edit/rezip the package or use the Chrome Ext (which isn't perfect due to the nature of the browser addon).

Regarding the implementation of the edit mode, I have no idea how to do that, I never code VS Ext.

I may help in some testing if required but not in coding.
